Checkout/Source/Tokenisation/CheckoutAPIService.swift

```diff
@@ -170,3 +171,53 @@ final public class CheckoutAPIService: CheckoutAPIProtocol {
     }
   }
 }
+
+// MARK: Security Code Request
+
+extension CheckoutAPIService {
+  public func createSecurityCodeToken(securityCode: String, completion: @escaping (Result<SecurityCodeTokenDetails, TokenisationError.SecurityCodeError>) -> Void) {
+    guard !publicKey.isEmpty else {
+        completion(.failure(.missingAPIKey))
+        return
+    }
+
+    let request = SecurityCodeRequest(tokenData: TokenData(securityCode: securityCode))
+    let requestParameterResult = requestFactory.create(request: .securityCodeToken(request: request, publicKey: publicKey))
+
+    switch requestParameterResult {
+    case .success(let requestParameters):
+//      logManager.queue(event: .tokenRequested(CheckoutLogEvent.TokenRequestData(
+//        tokenType: request.type,
+//        publicKey: publicKey
+//      )))
+      createSecurityCodeToken(requestParameters: requestParameters, completion: completion)
+    case .failure(let error):
+      switch error {
+      case .baseURLCouldNotBeConvertedToComponents, .couldNotBuildURL:
+        completion(.failure(.couldNotBuildURLForRequest))
+      }
+    }
+  }
```

Comment:
Before hitting the tokenisation API endpoint, do we validate the CVV SDK side and throw an error message to the user if they
1. Hit the pay button without adding CVV
2. If they inject AMEX and hit the pay button with 3 characters, should throw SDK-side internal error
3. If they inject no scheme (in this case unknown), hitting the pay button with less than 3 characters should throw SDK-side internal error

Reply:
1) returns false if a scheme is passed
2) yes, it returns false when the scheme is AMEX and there are 3 digits typed in
3) I'll need to add an extra check for the 0 digit case when there is no scheme. Currently it accepts `0, 3 and 4 digits`

@chintan-soni-cko thanks for the detailed review 🙂

Reply:
Thanks, Okhan. Sorry if misunderstood your answers. Are you referring here isEnteredCVVValid value or when hitting the tokenisation endpoint? In Android, there are no flags on whether it is valid or not while hitting the tokenisation. Just returning failure callback with an error message
<img width="347" alt="Screenshot 2023-10-11 at 15 34 27" src="https://github.com/checkout/frames-ios/assets/114917119/f4eec052-a313-4a68-8ebd-3b77f408cb7b">

Reply:
correct. it will be the same for iOS as in your screenshot @chintan-soni-cko

Reply:
as discussed offline, Android handles and returns error differently. On Android, there are error messages being returned, whereas in iOS we return error codes and the merchants' developers decide what to show as an error message. We'll keep the same approach.